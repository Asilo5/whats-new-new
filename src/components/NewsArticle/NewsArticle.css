.news-article {
    width: 40%;
    margin: 20px;
    padding: 10px;
    border-right: 1px solid gray;
    border-left: 1px solid gray;
}

.news-article > img{
    height: 24em;
    margin: 20px auto;
    display: block;
}