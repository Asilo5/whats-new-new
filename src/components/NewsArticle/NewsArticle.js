import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ headline, img, description, url }) => {
    return (
        <section className='news-article'>
            <img src={img} alt={headline} />
            <h2>{headline}</h2>
            <h3>{description}</h3>
            <a href={url}>Continue Reading Article</a>
        </section>
    )
}

export default NewsArticle;