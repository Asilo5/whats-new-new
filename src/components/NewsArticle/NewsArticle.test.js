import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match snapshot with all information passing in correctly', () => {
      let wrapper = shallow(<NewsArticle headline={'Title'}
                                         img={<img src='mockImage' />}
                                         description={'Blah to the Blah'}
                         />);

      expect(wrapper).toMatchSnapshot();
  })
})