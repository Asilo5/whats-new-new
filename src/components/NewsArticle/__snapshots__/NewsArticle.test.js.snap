// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match snapshot with all information passing in correctly 1`] = `
<section
  className="news-article"
>
  <img
    alt="Title"
    src={
      <img
        src="mockImage"
      />
    }
  />
  <h2>
    Title
  </h2>
  <h3>
    Blah to the Blah
  </h3>
  <a>
    Continue Reading Article
  </a>
</section>
`;
